test/test_sync_subscriptions.cpp

```diff
@@ -399,7 +399,9 @@ TEST(Sync_SubscriptionStoreRefreshSubscriptionSetInvalid)
     CHECK_THROW(latest->refresh(), RuntimeError);
 }
 
-TEST(Sync_SubscriptionStoreInternalSchemaMigration)
+// Only runs if REALM_MAX_BPNODE_SIZE is 1000 since that's what the pre-created
+// test realm files were created with
+TEST_IF(Sync_SubscriptionStoreInternalSchemaMigration, REALM_MAX_BPNODE_SIZE == 1000)
```

Comment:
Its too bad that we have to do this. Theoretically, we should support opening a Realm written with a different max node size because we know the size of every array, and we only need the max size to determine when to split while adding more items. But we have never claimed to support this and it isn't part of this PR so not a blocker for me.

Reply:
Yeah, I'm happy to revisit this, but these changes have lingered for a long time and I don't want to keep sitting on them to try to get these little changes perfect.